### PR TITLE
[TKW] Clear cache after every test for robustify test

### PR DESCRIPTION
**Pull request by raikonenfnu**
Tests on MI250 is randomly passing and failing. In order to robustify them, this PR adds manual_seed in places that are missing, and add cuda.clear_cache() to ensure we don't run out of memory.

### CODE REVIEW

File: tests/kernel/wave/wave_attention_test.py [hunk 1]
```diff
@@ -193,6 +193,8 @@ def repeat(
         schedule=enable_scheduling,
         use_scheduling_barriers=enable_scheduling_barriers,
     ):
+        torch.cuda.empty_cache()
```
Comment (harsh-nod) [inline tests/kernel/wave/wave_attention_test.py:196]: what do you think of moving this to a 
```
initialize_test(seed=0)
```
function in utils.py?

**Review by harsh-nod — CHANGES_REQUESTED** [review]
thanks! how can we validate that this is really the root cause and that it fixes the issue?

File: tests/kernel/wave/wave_attention_test.py (hunk 1 above)
Comment (raikonenfnu) [reply tests/kernel/wave/wave_attention_test.py:196]: as in a function that has both torch.cuda.empty_cache() + torch.manual_seed(seed)?

Comment (raikonenfnu) [reply tests/kernel/wave/wave_attention_test.py]: done, but renamed to `initialize_seed_and_cache` for better explanation to what it does. :)

**Pull request closed without merging**